### PR TITLE
clusters improvement #1 and #2

### PR DESCRIPTION
**Pull request by danielmarzini**
- decouple initial_node_count from node_count (cannot be set at the same time)
- using project_id as a parameters for the clusters variable to being able to place clusters into different projects

### CODE REVIEW

File: fast/stages/03-gke-multitenant/prod/main.tf [hunk 1]
```diff
@@ -16,13 +16,26 @@
...
-  _gke_robot_sa      = "serviceAccount:${module.gke-project-0.service_accounts.robots.container-engine}"
-  _cloud_services_sa = "serviceAccount:${module.gke-project-0.service_accounts.cloud_services}"
-  host_project_bindings = [
-    { role = "roles/container.hostServiceAgentUser", member = local._gke_robot_sa },
-    { role = "roles/compute.networkUser", member = local._gke_robot_sa },
-    { role = "roles/compute.networkUser", member = local._cloud_services_sa }
+  
+  _gke_robot_sas      = [
+    "serviceAccount:${module.gke-project-0.service_accounts.robots.container-engine}",
+    "serviceAccount:${module.gke-project-1.service_accounts.robots.container-engine}"
```
Comment (juliocc) [inline fast/stages/03-gke-multitenant/prod/main.tf:22]: Managing bindings for more than one project unnecessarily complex as this file is supposed to have a single project. The `gke-project-1` was a test I left behind accidentally. Please rebase your branch

**Review by juliocc — CHANGES_REQUESTED** [review]
Hey @danielmarzini. Thanks for the PR, se comments below:

Yu have to rebase your branch. Once you do that you'll notice you don't have to do the thing you're doing with bindings in the host project.

The part regarding the project_id seems a little redundant since we're deploying a single project (at least initially) but we can keep it.

The initial_node_count fix is definitely needed since it's currently broken.

File: fast/stages/03-gke-multitenant/prod/main.tf (hunk 1 above)
Comment (danielmarzini) [reply fast/stages/03-gke-multitenant/prod/main.tf:22]: @juliocc I re-added the project-1, I think having a two projects scenario brings value for a multitenant setup, but, obviously, let's discuss about this.

Not sure about the things that I did with the binding, I did them cause weren't managed somewhere else, where the point where they are already managed?

File: fast/stages/03-gke-multitenant/prod/main.tf [hunk 2]
```diff
@@ -66,15 +74,62 @@ module "gke-project-0" {
   # }
 }
 
+module "gke-project-1" {
```
Comment (juliocc) [inline fast/stages/03-gke-multitenant/prod/main.tf:77]: I prefer having a single project, at least initially. As you can see things get more complicated if we allow more than one.

**Review by juliocc — CHANGES_REQUESTED** [review]

**Comment by danielmarzini** [comment]
@juliocc remove gke-project-1 and its binding, left there the structure to host future projects addition. wdyt?

File: fast/stages/03-gke-multitenant/prod/main.tf [hunk 3]
```diff
@@ -73,8 +79,14 @@ module "gke-dataset-resource-usage" {
   friendly_name = "GKE resource usage."
 }
 
-resource "google_project_iam_member" "host_project_bindings" {
-  for_each = { for i, v in local.host_project_bindings : i => v }
+resource "google_project_iam_member" "host_project_gke_robot_bindings" {
```
Comment (juliocc) [inline fast/stages/03-gke-multitenant/prod/main.tf:82]: why split these in two? It easier to manage the depends_on that you need on the cluster if you have a single resource.

File: fast/stages/03-gke-multitenant/prod/main.tf [hunk 4]
```diff
@@ -16,13 +16,19 @@
...
-    { role = "roles/compute.networkUser", member = local._gke_robot_sa },
-    { role = "roles/compute.networkUser", member = local._cloud_services_sa }
+  
+  _gke_robot_sas      = [
+    "serviceAccount:${module.gke-project-0.service_accounts.robots.container-engine}"
+  ]
+  _cloud_services_sas = [
+    "serviceAccount:${module.gke-project-0.service_accounts.cloud_services}"
+  ]
+  host_project_cloud_services_bindings = [ for member in local._cloud_services_sas :
```
Comment (juliocc) [inline fast/stages/03-gke-multitenant/prod/main.tf:26]: why do you split this in two variables? Can't you just create variable as before?

**Review by juliocc — CHANGES_REQUESTED** [review]
left a few comments on the code. Also, can you remove the intermediary commits (or at least squash when mergin back to fast-dev-gke)?

**Comment by danielmarzini** [comment]
@juliocc now adjusted

**Review by juliocc — APPROVED** [review]

**Pull request merged**